src/Components/Forms/src/DataAnnotationsValidator.cs

```diff
@@ -22,7 +24,20 @@ protected override void OnInitialized()
                     $"inside an EditForm.");
             }
 
-            CurrentEditContext.AddDataAnnotationsValidation();
+            _subscriptions = CurrentEditContext.EnableDataAnnotationsValidation();
```

Comment:
Do you need to account for the `EditContext` instance changing between these calls?

Reply:
Not sure what you mean. This is inside `OnInitialized`, so there's only one call to it per component instance.

Reply:
Something like this: https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/InputBase.cs#L218-L227

Reply:
Oh, so is your question something like "do we need to support changing `EditContext`"? If so I think the answer is no, or at least it would be unrelated to this PR. The `<DataAnnotationsValidator />` component already ignores any attempt to change the `EditContext` cascaded parameter, and this PR doesn't change that.

Reply:
I guess more along the lines of should we fail loudly if the EditContext changes. I guess the PR slightly aggravates things because there's a memory leak now if the EditContext changes.

Reply:
Is there a memory leak? Each `DataAnnotationsValidator` instance is associated with exactly one `EditContext`, which is determined at initialization. The `DataAnnotationsValidator`'s disposal always unsubscribes it from that `EditContext`. The only way memory gets leaked is if you don't dispose the `DataAnnotationsValidator`, isn't it?

Note that there would have been a possible leaky pattern before where you keep destroying and recreating a `DataAnnotationsValidator`, and on each cycle you'd add another subscription to the `EditContext` which never gets released. As of this PR that's no longer leaky because each time a `DataAnnotationsValidator` gets removed from the UI, its subscriptions will automatically be released.

Reply:
Yup. Not a memory leak since a newer EditContext instance simply get ignored. 👍🏽

Reply:
Anyway, I agree it would be better to have an exception rather than silently ignoring the fact that we don't support changing the edit context, so I've added that. Let me know if you have other concerns.